querybook/webapp/components/DataDocDAGExporter/DataDocDAGExporterSettings.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from 'react';
+
+import { titleize } from 'lib/utils';
+import { useExporterSettings } from 'hooks/dag/useExporterSettings';
+
+import { SmartForm } from 'ui/SmartForm/SmartForm';
+import { SimpleReactSelect } from 'ui/SimpleReactSelect/SimpleReactSelect';
+import { FormSectionHeader } from 'ui/Form/FormField';
+import { Button } from 'ui/Button/Button';
+import { AsyncButton } from 'ui/AsyncButton/AsyncButton';
+import { CopyPasteModal } from 'ui/CopyPasteModal/CopyPasteModal';
+import { Modal } from 'ui/Modal/Modal';
+
+interface IProps {
+    onExport: (name: string, settings: any) => Promise<string>;
+    savedMeta: Record<string, any>;
+    onSave: (meta: any) => Promise<any>;
+    clearExportData: () => void;
+    exportData?: string;
+}
+
+export const DataDocDAGExporterSettings: React.FunctionComponent<IProps> = ({
+    onExport,
+    savedMeta,
+    onSave,
+    clearExportData,
+    exportData,
+}) => {
+    const {
+        exporterNames,
+        selectedExporter,
+        setSelectedExporter,
+        settingValues,
+        exporterMeta,
+        handleSettingValuesChange,
+        exportType,
+    } = useExporterSettings({ savedMeta });
+
+    const exportModalTitle = React.useMemo(
+        () => `Export to ${titleize(selectedExporter, '_', ' ')}`,
+        [selectedExporter]
+    );
+
+    React.useEffect(() => {
+        clearExportData();
```

Comment:
i guess this hook is unnecessary (when the modal is dismissed, the exportData is removed)